bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/layout/NanoleafLayout.java

```diff
@@ -0,0 +1,181 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.imageio.ImageIO;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.nanoleaf.internal.NanoleafBindingConstants;
+import org.openhab.binding.nanoleaf.internal.layout.shape.Shape;
+import org.openhab.binding.nanoleaf.internal.layout.shape.ShapeFactory;
+import org.openhab.binding.nanoleaf.internal.model.GlobalOrientation;
+import org.openhab.binding.nanoleaf.internal.model.Layout;
+import org.openhab.binding.nanoleaf.internal.model.PanelLayout;
+import org.openhab.binding.nanoleaf.internal.model.PositionDatum;
+
+/**
+ * Renders the Nonoleaf layout to an image.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class NanoleafLayout {
+
+    private static final Color COLOR_BACKGROUND = Color.WHITE;
+    private static final Color COLOR_PANEL = Color.BLACK;
+    private static final Color COLOR_SIDE = Color.GRAY;
+    private static final Color COLOR_TEXT = Color.BLACK;
+
+    public static byte[] render(PanelLayout panelLayout) throws IOException {
+        double rotationRadians = 0;
+        GlobalOrientation globalOrientation = panelLayout.getGlobalOrientation();
+        if (globalOrientation != null) {
+            rotationRadians = calculateRotation(globalOrientation);
+        }
+
+        Layout layout = panelLayout.getLayout();
+        if (layout == null) {
+            return new byte[] {};
+        }
+
+        List<PositionDatum> panels = layout.getPositionData();
+        if (panels == null) {
+            return new byte[] {};
+        }
+
+        Point2D size[] = findSize(panels, rotationRadians);
+        final Point2D min = size[0];
+        final Point2D max = size[1];
+        Point2D prev = null;
+        Point2D first = null;
+
+        int sideCounter = 0;
+        BufferedImage image = new BufferedImage(
+                (max.getX() - min.getX()) + 2 * NanoleafBindingConstants.LAYOUT_BORDER_WIDTH,
+                (max.getY() - min.getY()) + 2 * NanoleafBindingConstants.LAYOUT_BORDER_WIDTH,
+                BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2 = image.createGraphics();
+
+        g2.setBackground(COLOR_BACKGROUND);
+        g2.clearRect(0, 0, image.getWidth(), image.getHeight());
+
+        for (PositionDatum panel : panels) {
+            final ShapeType shapeType = ShapeType.valueOf(panel.getShapeType());
+
+            Shape shape = ShapeFactory.CreateShape(shapeType, panel);
+            List<Point2D> outline = toPictureLayout(shape.generateOutline(), image.getHeight(), min, rotationRadians);
+            for (int i = 0; i < outline.size(); i++) {
+                g2.setColor(COLOR_SIDE);
+                Point2D pos = outline.get(i);
+                Point2D nextPos = outline.get((i + 1) % outline.size());
+                g2.drawLine(pos.getX(), pos.getY(), nextPos.getX(), nextPos.getY());
```

Comment:
Would it be possible to use draw with a polygon shape, so in the future we could even fill that shape with a particular color? (just asking ...)

Reply:
Ive been thinking a little about color. I think for that we should add another channel which has color but not the thing IDs(?) For the shapes it is pretty easy, but for the elements where each corner has an individual light it is more interesting because you would have to blend colors in triangles - one for each side of the hexagon - so I dont think we will draw polygons then either, or maybe 6 of them with separate Paints.